File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/api/messages/multidevice/RegisterAsSecondaryDeviceResponse.java
package org.whispersystems.signalservice.api.messages.multidevice;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.UUID;

public class RegisterAsSecondaryDeviceResponse {
  @JsonProperty
  private UUID uuid;

  @JsonProperty
  private UUID pni;

  @JsonProperty
  private String deviceId;

  public RegisterAsSecondaryDeviceResponse() {}

  public RegisterAsSecondaryDeviceResponse(UUID uuid, UUID pni, String deviceId) {
    this.uuid     = uuid;
    this.pni      = pni;
    this.deviceId = deviceId;
  }

  public UUID getUuid() {
    return uuid;
  }

  public UUID getPni() {
    return pni;
  }

  public String getDeviceId() {
    return deviceId;
  }
}

File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/api/messages/multidevice/RegisterAsSecondaryDeviceResponse.kt
/*
 * Copyright 2025 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.api.messages.multidevice

import java.util.UUID

class RegisterAsSecondaryDeviceResponse(
  val uuid: UUID,
  val pni: UUID,
  val deviceId: String
)
